packages/backend/error/src/api-error.ts

```diff
@@ -1,99 +1,50 @@
 import { createDebug } from "@gravitywelluk/debug";
 
-const debug = createDebug("ERROR");
+import { ErrorType } from "./error-type";
 
-/** Available error codes */
-export enum ErrorType {
-  ApiError = "API_ERROR",
-  ApiConectionError = "API_CONNECTION_ERROR",
-  AuthenticationError = "AUTHENTICATION_ERROR",
-  ForbiddenError = "FORBIDDEN_ERROR",
-  InvalidData = "INVALID_DATA_ERROR",
-  DatabaseError = "DATABASE_ERROR",
-  NotFoundError = "NOT_FOUND_ERROR",
-  ThirdPartyError = "THIRD_PARTY_ERROR",
-  TooManyRequests = "TOO_MANY_REQUESTS",
-  UnknownError = "UNKNOWN_ERROR"
-}
+const debug = createDebug("ERROR");
 
-export interface ApiErrorResponse<C = unknown> {
-  /** Api response code */
-  statusCode: number;
-  /** Generic error type */
-  type: ErrorType;
+export interface ApiErrorResponse {
+  /** HTTP status code */
+  statusCode: ErrorType;
+  /** Title of HTTP status code */
+  title: string;
+  /** User-facing error message */
   message: string;
-  /** More specific error code, types specified from parent */
-  code?: C;
-  /** Extra more specific relevant information */
-  param?: Record<string, unknown>;
+  /** Additional relevant error details */
```

Comment:
detail feels a bit of a duplication of message and quite specific. I dont think it adds any functionality. I think if we bother having a third parameter it would be more loosely typed than this. I like the generic on `code` was good actually  but I think it just shouldn't be labelled as Code as thats very specific. We could label the generic as context. This way you can adapt the error class however you like. Context is non-specific so could be a string or could be an object with specific useful data to that error type etc.